leapp/utils/workarounds/fqdn.py

```diff
@@ -0,0 +1,18 @@
+import re
+import socket
+
+
+def decorate_getfqdn(fn):
+    def wrap(*args, **kwargs):
+        result = fn(*args, **kwargs)
```

Comment:
I think we should be careful here about the string type. In the case of the reported problem there were unicode characters in the hostname, so in that case `socket.getfqdn` might not return an instance of `str` but an instance of `unicode` @ python 2.7 - In that case there might be a problem with this code.

Also if something like this happens on python 3 it actually might instead return bytes, we need to validate that I guess.

WDYT?

Reply:
I just tested this and in both cases `str` was returned.

Reply:
Added some tests